device/edison.go

```diff
@@ -36,75 +37,169 @@ const (
+		args := []string{
+			fmt.Sprintf("%s@%s", vbox.VBoxUser, vbox.VBoxIP),
+			"-p",
+			vbox.VBoxSSHPort,
+			constants.TMP_DIR + script,
 		}
+		if err := help.ExecStandardStd("ssh", args...); err != nil {
+			fmt.Println("[-] Can't find Intel Edison board, please try to re-connect it")
 
-		if err := vbox.Stop(d.vbox.UUID); err != nil {
-			log.Error(err)
+			if !dialogs.YesNoDialog("Type yes once connected.") {
+				fmt.Println("Exiting with exit status 2 ...")
+				os.Exit(2)
+			}
+			continue
 		}
+		break
+	}
 
-		job := help.NewBackgroundJob()
-		go func() {
-			defer job.Close()
-			time.Sleep(120 * time.Second)
-		}()
+	if err := vbox.Stop(d.vbox.UUID); err != nil {
+		log.Error(err)
+	}
+
+	job := help.NewBackgroundJob()
+	go func() {
+		defer job.Close()
+		time.Sleep(120 * time.Second)
+	}()
+
+	help.WaitJobAndSpin("Your Edison board is restarting...", job)
+	return nil
+}
 
-		help.WaitJobAndSpin("Configuring Edison", job)
+func (d *edison) flashWindows() error {
+	fileName := ""
+	filePath := ""
+	if fn, fp, err := d.flasher.DownloadImage(); err == nil {
+		fileName = fn
+		filePath = fp
+	} else {
+		return err
+	}
+
+	fmt.Printf("[+] Extracting %s \n", fileName)
+	if !strings.HasSuffix(fileName, ".zip") {
+		return nil
+	}
+	extractedPath := help.GetTempDir() + help.Separator() + strings.TrimSuffix(fileName, ".zip") + help.Separator()
+	command := "unzip"
+	args := []string{
+		"-o",
+		filePath,
+		"-d",
+		extractedPath,
+	}
+	log.WithField("args", args).Debug("Extracting an image...")
+	if out, err := exec.Command(command, args...).CombinedOutput(); err != nil {
+		log.WithField("out", out).Error(err)
+		fmt.Println("[-] Error extracting image!", out)
+		return err
+	}
+
+	d.getDFUUtil(extractedPath)
+
+	script := extractedPath + help.Separator() + "flashall.bat"
+	log.Debug("Running flashall... ", script)
+	cmd := exec.Command(script)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Stdin = os.Stdin
+	cmd.Dir = extractedPath
+	if err := cmd.Run(); err != nil {
+		fmt.Println("[-] Can't find Intel Edison board, please try to re-connect it")
+		os.Exit(2)
 	}
+	job := help.NewBackgroundJob()
+	go func() {
+		defer job.Close()
+		time.Sleep(120 * time.Second)
+	}()
 
+	help.WaitJobAndSpin("Your Edison board is restarting...", job)
+	return nil
+}
+
+func (d *edison) getDFUUtil(dst string) error {
+	url := "https://cdn.isaax.io/isaax-distro/utilities/dfu-util/dfu-util-0.9-win64.zip"
+
+	if help.Exists(dst + "dfu-util.exe") {
+		return nil
+	}
+
+	wg := &sync.WaitGroup{}
+	fileName, bar, err := help.DownloadFromUrlWithAttemptsAsync(url, dst, 5, wg)
+	if err != nil {
+		return err
+	}
+
+	bar.Prefix(fmt.Sprintf("[+] Download %-15s", fileName))
+	bar.Start()
+	wg.Wait()
+	bar.Finish()
+	time.Sleep(time.Second)
+
+	log.WithField("dst", dst).Debug("Extracting")
+	if out, err := exec.Command("unzip", "-j", "-o", dst+"dfu-util-0.9-win64.zip", "-d", dst).CombinedOutput(); err != nil {
+		return err
+	} else {
```

Comment:
__[golint]__
* [golint] if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)

<!-- Comment by [SideCI](https://sideci.com) -->